lib/views/pay/pay_view.dart

```diff
@@ -127,8 +129,32 @@ class _PayViewState extends State<PayView> {
                 setState(() {
                   _invoiceController.text = data!.text!;
                 });
-                boltString = data!.text;
-                invoiceActions(data.text!);
+                if (data!.text!.startsWith('ln')) {
+                  /// This is a lightning invoice
+                  boltString = data.text;
+                  invoiceActions(boltString!);
+                } else if (data.text!.startsWith('tb1')) {
+                  /// This is a btc address
+                  btcAddress = data.text;
+                  if (context.mounted) {
+                    Navigator.of(context).push(MaterialPageRoute(
+                        builder: (context) => NumberPad(
+                              provider: widget.provider,
+                              btcAddress: btcAddress,
+                            )));
+                  }
```

Comment:
```suggestion
                } else if (data.text!.startsWith('tb1')) {
                  /// This is a btc address
                  btcAddress = data.text;
                  if (context.mounted) {
                    Navigator.of(context).push(MaterialPageRoute(
                        builder: (context) => NumberPad(
                              provider: widget.provider,
                              btcAddress: btcAddress,
                            )));
                  }
```

`tb1` is only one of many bitcoin addresses, try to search how many address in bitcoin there are per network. This match does not work. Maybe you can assume that if this does not start with `ln` is always a bitcoin address but this is broken too